Converters/Tekla/Speckle.Converter.Tekla2024/ToSpeckle/Raw/BeamRawConverter.cs

```diff
@@ -0,0 +1,58 @@
+using Speckle.Converters.Common;
+using Speckle.Converters.Common.Objects;
+using Speckle.Sdk.Models;
+using SOG = Speckle.Objects.Geometry;
+
+namespace Speckle.Converter.Tekla2024.ToSpeckle.Raw;
+
+public class BeamRawConverter : ITypedConverter<TSM.Beam, Base>
+{
+  private readonly IConverterSettingsStore<TeklaConversionSettings> _settingsStore;
+  private readonly ITypedConverter<TG.Point, SOG.Point> _pointConverter;
+  private readonly ITypedConverter<TSM.Solid, SOG.Mesh> _meshConverter;
+
+  public BeamRawConverter(
+    IConverterSettingsStore<TeklaConversionSettings> settingsStore,
+    ITypedConverter<TG.Point, SOG.Point> pointConverter,
+    ITypedConverter<TSM.Solid, SOG.Mesh> meshConverter
+  )
+  {
+    _settingsStore = settingsStore;
+    _pointConverter = pointConverter;
+    _meshConverter = meshConverter;
+  }
+
+  public Base Convert(TSM.Beam target)
+  {
+    var beamObject = new Base
+    {
+      ["type"] = nameof(TSM.Beam),
+      ["units"] = _settingsStore.Current.SpeckleUnits,
+      ["profile"] = target.Profile.ProfileString,
+      ["material"] = target.Material.MaterialString,
+    };
+
+    var solid = target.GetSolid();
+    var mesh = _meshConverter.Convert(solid);
+
+    var color = new TSMUI.Color();
+    TSMUI.ModelObjectVisualization.GetRepresentation(target, ref color);
+
+    int r = (int)(color.Red * 255);
+    int g = (int)(color.Green * 255);
+    int b = (int)(color.Blue * 255);
+    int argb = (255 << 24) | (r << 16) | (g << 8) | b;
+
+    int vertexCount = mesh.vertices.Count / 3;
+
+    mesh.colors = new List<int>(vertexCount);
+    for (int i = 0; i < vertexCount; i++)
+    {
+      mesh.colors.Add(argb);
+    }
```

Comment:
thats good for validation of argb value but should be removed. We will handle them with Proxies, otherwise it won't work with other apps.

OR: you can comment out to remove later by noting "it will be replaced with proxy implementation" bla bla